### ISSUE TITLE
Try Monotonic Attention

### ISSUE THREAD

**Issue by lifeiteng** [keithito/tacotron#24]
I tried Monotonic attention and got better result (alignment is clear at step 10k).
![eval-125000-0](https://user-images.githubusercontent.com/3107839/29455089-5d1f980e-8442-11e7-91fd-3116be6763ae.png)

1. update to TensorFlow 1.3
2. modify models/tacotron.py `BahdanauAttention(256, encoder_outputs) -> BahdanauMonotonicAttention(256, encoder_outputs)`

[eval-125k.zip](https://github.com/keithito/tacotron/files/1234092/eval-125k.zip)

**Comment by DarkDefender** [comment]
Nice! I wonder if it will sound much better than the current 877k result.
IE will the end quality converge with the previous method or will it be able to surpass it.

The results ar 125k are already really nice. @lifeiteng do you plan to train it further?

**Comment by ElevenGameStudios** [comment]
@lifeiteng I agree that the convergence time and attention alignment are really impressive..
But I am not quite sure about the voice, it sounds pretty clear, but also weirdly stretched out at this point. I wonder whether it will converge to a more natural sounding rhythm over training time? Or is this kind of rhythmic "normalization" part of the monotonic attention?
Well, yeah, it would be really interesting to see how this converges after more training!

**Comment by lifeiteng** [comment]
`Yuxuan` the first author of Tacotron said that they also use Monotonic Attention in their newest version. They showed `paragraph  synthesis(more than 400 chars)`.

**Comment by candlewill** [comment]
@lifeiteng Do you have the slides of Tacotron on Interspeech?

**Comment by lifeiteng** [comment]
@candlewill It's poster, not oral.

**Comment by DarkDefender** [comment]
@lifeiteng Did you think that the paragraph version sounded more natural/better? If I took a wild guess, I would suspect that it would introduce a better flow between sentences in a paragraph. But not much else.

**Comment by lifeiteng** [comment]
@DarkDefender The audios' quality is great, but I did not make a careful comparison, even I think all examples are the newest version.
![wechatimg7](https://user-images.githubusercontent.com/3107839/29729915-66512cd4-89de-11e7-8d57-80e25e065fcc.jpeg)

**Comment by DarkDefender** [comment]
@lifeiteng Did they post the updated samples anywhere public? I have been able to find any official updates.

Besides that, have you managed to train the Monotonic Attention model any further? As we discussed previously it would be interesting to see if the quality improves dramatically over the current non-monotonic speech samples.

**Comment by lifeiteng** [comment]
1. It seems not published.
2. Actually I failed to reproduce the mono-attention result on the newest version of this repo.

But, I got pretty cool result (alignment is clear at  6k step,  including Single and Multi-Speaker datasets) on my own implementation (different audio feature processing and other tunings) using mono-attention. Multi-Speaker version is same as Deep Voice2.

**Comment by pineking** [comment]
@lifeiteng could the different feature process show the significant improvement?

**Comment by lifeiteng** [comment]
@pineking I test pcm value range [-1, 1] and  [-32767， 32768]，[-1, 1] fail to converge (this is strange, because FFT(x*32768) = 32768 * FFT(x))，the first author of Tacotron told me they use [-1, 1]. I don't speed time on this, just one pair experiments.

**Comment by Saxenauts** [comment]
@lifeiteng Hey, could you share results, if you trained it further, and if it converged?  (After 125k steps, that is). 
Also, what dataset are you using? 
And, as @ElevenGameStudios pointed out, the syllables seems stretched out. Is it because of monotonic attention? 
Thanks.

**Comment by lifeiteng** [comment]
## results after 125k (This repo)
[step-550k.zip](https://github.com/keithito/tacotron/files/1305037/step-550k.zip)
[step-883k.zip](https://github.com/keithito/tacotron/files/1305038/step-883k.zip)

## multi-speaker results (my implementation)
[read.zip](https://github.com/keithito/tacotron/files/1305049/read.zip)
```
He has read the whole thing.
He reads books.
```

**Comment by lifeiteng** [comment]
close it.

**Comment by dy-octa** [comment]
Hi~Have you ever used any other tricks in training? When I simply substitute the attention mechanism with the monotonic one, the model would fail to attend to the correct encoder step during inference (without ground truth given), but rapidly reach the last encoder step in a few steps, though the model would work when ground truth aligned.
